scripts/fix_doc_ipfs_hashes.py

```diff
@@ -35,25 +39,138 @@ def read_file(filepath: str) -> str:
+            self.name,
+            f"{'aea-config' if self.type == 'agent' else self.type}.yaml",
+        )
+        with open(yaml_file_path, "r", encoding="utf-8") as file:
+            content = yaml.load_all(file, Loader=yaml.FullLoader)
+            for resource in content:
+                if "version" in resource:
+                    self.last_version = resource["version"]
+                    break
+
+    def get_command(self) -> str:
+        """Get the add command"""
+        if self.type == "agent":
+            return (
+                f"aea fetch {self.vendor}/{self.name}:{self.last_version}:{self.hash}"
+            )
+        return f"aea add {self.type} {self.vendor}/{self.name}:{self.last_version}:{self.hash}"
+
+
+class PackageHashManager:
+    """Class that represents the packages in hashes.csv"""
+
+    def __init__(self) -> None:
+        """Constructor"""
+        hashes_file = Path("packages", "hashes.csv").relative_to(".")
+        with open(hashes_file, "r", encoding="utf-8") as file_:
+            self.packages = [Package(line) for line in file_.readlines()]
+            self.packages = [p for p in self.packages if p.name != "scaffold"]
+
+        self.package_tree: Dict = {}
+        for p in self.packages:
+            self.package_tree.setdefault(p.vendor, {})
+            self.package_tree[p.vendor].setdefault(p.type, {})
+            self.package_tree[p.vendor][p.type].setdefault(p.name, p)
+
+    def get_package_by_hash(self, package_hash: str) -> Optional[Package]:
+        """Get a package given its hash"""
+        packages = list(filter(lambda p: p.hash == package_hash, self.packages))
+        if not packages:
+            return None
+        if len(packages) > 1:
+            raise ValueError(
+                f"PackageHashManager: hash search for {package_hash} returned more than 1 result in hashes.csv"
+            )
+        return packages[0]
+
+    def get_hash_by_package_line(
+        self, package_line: str, md_file: str
+    ) -> Optional[str]:
+        """Get a hash given its package line"""
+
+        try:
+            m = re.match(AEA_COMMAND_REGEX, package_line)
+
+            # No match
+            if not m:
+                print(
+                    f"Docs [{md_file}]: line '{package_line}' does not match an aea command format"
+                )
+                return None
+            d = m.groupdict()
+
+            # Underspecified commands that only use the hash
+            if not d["vendor"] and not d["package"]:
+                package = self.get_package_by_hash(d["hash"])
+
+                # This hash exists in hashes.csv
+                if package:
+                    return package.hash
+
+                # This hash does not exist in hashes.csv
+                print(
+                    f"Docs [{md_file}]: unknown IPFS hash in line '{package_line}'. Can't fix because this command just uses the hash"
+                )
+                return None
```

Comment:
should we not raise an error here? We don't want IPFS hashing floating around in docs that we cannot match I think

Reply:
We do, but only we have checked all files: we return None here and then in `fix_ipfs_hashes`:
```
            if not expected_hash:
                errors = True
                continue
            expected_package = package_manager.get_package_by_hash(expected_hash)
            if not expected_package:
                errors = True
                continue

    ...

    if errors:
        raise ValueError(
            "There were some errors while processing the docs. Check the logs."
        )
```